PresentScreenings/Controllers/GoToScreeningDialog.cs

```diff
@@ -10,10 +12,80 @@ namespace PresentScreenings.TableView
 
     public abstract class GoToScreeningDialog : NSViewController
     {
+        #region Constants
+        const float _yBetweenLabels = ControlsFactory.VerticalPixelsBetweenLabels;
+        const float _xBetweenLabels = _yBetweenLabels;
+        const float _labelHeight = ControlsFactory.StandardLabelHeight;
+        const float _buttonWidth = _labelHeight;
+        #endregion
+
+        #region Application Access
+        static AppDelegate _app => (AppDelegate)NSApplication.SharedApplication.Delegate;
+        #endregion
+
+        #region Private Members
+        static Dictionary<Screening, NSTextField> _labelByfilmScreening;
+        #endregion
+
+        #region Constructors
         public GoToScreeningDialog(IntPtr handle) : base(handle)
         {
         }
+        #endregion
+
+        #region Virtual Methods
+        public static void DisplayScreeningControls(List<Screening> screenings, NSView screeningsView,
+            GoToScreeningDelegate goToScreening, ref FilmScreeningControl currentScreeningControl)
+        {
+            _labelByfilmScreening = new Dictionary<Screening, NSTextField> { };
+            var yScreening = screeningsView.Frame.Height;
+            var contentWidth = screeningsView.Frame.Width;
+            foreach (var screening in screenings)
+            {
+                float xScreening = 0;
+                yScreening -= _labelHeight;
+
+                // Create the screening info button.
+                var buttonRect = new CGRect(xScreening, yScreening, _buttonWidth, _labelHeight);
```

Comment:
Why not create one frame rect before the loop and only change it Y-value before using it for a new control?
Just like the Friend controls in class `ScreeningDialogController`.

Reply:
Done.